src/main/java/th/ac/kmitl/science/comsci/example/nameprinters/NamePrinterLoader.java

```diff
@@ -0,0 +1,56 @@
+package th.ac.kmitl.science.comsci.example.nameprinters;
+
+import org.reflections.Reflections;
+
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+
+public class NamePrinterLoader implements Iterable<NamePrinter> {
+
+    //Singleton implementation
+    private static NamePrinterLoader instance;
+    private final String packageIdentifier;
+    private List<NamePrinter> namePrinters;
+    private NamePrinterLoader() {
+        packageIdentifier = getClass().getPackage().getName();
+
+        try {
+            namePrinters = getAllNamePrinters();
+        } catch (Exception e) {
```

Comment:
Why we need to catch all exception, please identify?

Reply:
I forgot to remove it after changing reflection-related code to use org.reflection library. An old code use some method from Reflection API which is throws Exception so I have no choice because it's unpredictable. Without exception catching should be safe now so I'll remove it.

Reply:
Kindly find the specific exception that need to catch instead of catching the whole exception.